packages/e2e-test-utils-playwright/src/site-editor/toggle-canvas-mode.js

```diff
@@ -4,5 +4,7 @@
  * @this {import('.').SiteEditor}
  */
 export async function enterEditMode() {
-	await this.page.click( '.edit-site-site-hub__edit-button' );
+	await this.page.click(
+		'.edit-site-visual-editor__editor-canvas[role="button"]'
+	);
```

Comment:
Using CSS selectors is not recommended. Prefer [`getByRole`](https://playwright.dev/docs/locators#locate-by-role) instead.

```suggestion
	await this.page
		.getByRole( 'button', { name: 'Edit' } )
		.click();
```

(We might want to make this button more explicit with additional text or region for screen readers though.)

In addition, not something we need to do in this PR, but let's not create utils with only one or two lines if they can be [inlined](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/README.md#dont-overload-test-utils-inline-simple-utils).

Reply:
Where does the "Edit" name come from, I don't think this has an "edit" name. It has an "Editor canvas" title instead. (It's an iframe that behaves like a button)

Reply:
Isn't there still an "Edit" button in the sidebar? But if we want to target the iframe we can do:

```suggestion
	await this.page
		.getByRole( 'button', { name: 'Editor canvas' } )
		.click();
```

Reply:
> Isn't there still an "Edit" button in the sidebar? 

No, not always, it's only present if your on some specific sidebars (single items).

Reply:
It turns out neither of these work because we need to click inside the canvas instead. something like `canvas().click( 'body' )`

Reply:
It seems to work if we specify the position explicitly:

```js
await page
	.getByRole( 'button', { name: 'Editor canvas' } )
	.click( { position: { x: 10, y: 10 } } );
```

By default, Playwright will find an arbitrary position, usually the center, to click on. But sometimes clicking on the center won't enter edit mode if the canvas is not full height. For instance, clicking on the bottom half of the canvas when viewing a small footer template part won't enter edit mode. This seems like a bug, is it?

(I guess specifying `{ x: 0, y: 0 }` also won't work because the canvas has `border-radius`.)

Reply:
> By default, Playwright will find an arbitrary position, usually the center, to click on. But sometimes clicking on the center won't enter edit mode if the canvas is not full height. For instance, clicking on the bottom half of the canvas when viewing a small footer template part won't enter edit mode. This seems like a bug, is it?

Yeah, it's a bug I think but it's outside the scope of this PR. For now I removed the utility like you suggested and just went with canvas.